network/socket-bench-server/src/lib.rs

```diff
@@ -96,22 +97,32 @@ pub fn build_tcp_noise_transport() -> impl Transport<Output = NoiseSocket<TcpSoc
 
 /// Server side handler for send throughput benchmark when the messages are sent
 /// over a simple stream (tcp or in-memory).
-pub async fn server_stream_handler<L, I, S, E>(mut server_listener: L)
+pub async fn server_stream_handler<L, I, S, E>(server_listener: L, executor: Handle)
 where
     L: Stream<Item = Result<(I, NetworkAddress), E>> + Unpin,
     I: Future<Output = Result<S, E>>,
-    S: AsyncRead + AsyncWrite + Unpin,
+    S: AsyncRead + AsyncWrite + Unpin + Send + 'static,
     E: ::std::error::Error,
 {
     // Wait for next inbound connection
-    while let Some(Ok((f_stream, _client_addr))) = server_listener.next().await {
-        let stream = f_stream.await.unwrap();
-        let mut stream = Framed::new(IoCompat::new(stream), LengthDelimitedCodec::new());
+    server_listener
+        .for_each_concurrent(1024, |result| async {
+            match result {
+                Ok((f_stream, _)) => {
+                    let stream = f_stream.await.unwrap();
+                    let mut stream =
+                        Framed::new(IoCompat::new(stream), LengthDelimitedCodec::new());
 
-        // Drain all messages from the client.
-        while let Some(_) = stream.next().await {}
-        stream.close().await.unwrap();
-    }
+                    // Drain all messages from the client.
+                    executor.spawn(async move {
+                        while let Some(_) = stream.next().await {}
+                        stream.close().await.unwrap();
+                    });
```

Comment:
i think this would make more sense like:

```rust
let bounded_executor = BoundedExecutor::new(128, executor);

// it's probably ok to just loop here w/o a task queue, since handling
// new tcp connection should not be bottleneck here
while let Some(Ok(f_stream. _)) = server_listener.next().await {
    // instead, we want to make sure we're spawning a task that includes the
    // connection upgrade (what we're measuring in this PR). this is because
    // spawned tasks (as opposed to `FuturesUnordered` in `Stream::for_each_concurrent`)
    // can actually run on multiple threads. the code before was running all
    // handshakes on one thread.
    bounded_executor.spawn(async move {
        // this will await the connection upgrade (i.e., noise handshake)
        let stream = f_stream.await.unwrap();
        let mut stream = Framed::new(IoCompat::new(stream), LengthDelimitedCodec::new());

        // drain all messages from client
        while let Some(_) = stream.next().await {}
        stream.close().await.unwrap();
    )}.await; // backpressure (could also try_spawn to just drop connections if at capacity)
}
```

Reply:
we should simulate the current implementation which is for_each_concurrent